src/main/java/com/gregtechceu/gtceu/syncsystem/data_transformers/collections/ListTransformer.java

```diff
@@ -30,8 +31,16 @@ public Tag serializeNBT(List<T> value, ISyncManaged holder) {
     public List<T> deserializeNBT(Tag tag, ISyncManaged holder, List<T> current) {
         if (!(tag instanceof ListTag listTag) || elementTransformer == null) return List.of();
 
-        return listTag.stream()
-                .map((t) -> elementTransformer.deserializeNBT(IValueTransformer.stripLdlibWrapper(t), null, null))
-                .toList();
+        try {
+            current.clear();
+            current.addAll(listTag.stream()
+                    .map((t) -> elementTransformer.deserializeNBT(IValueTransformer.stripLdlibWrapper(t), null, null))
+                    .toList());
```

Comment:
```suggestion
            listTag.stream()
                    .map(t -> elementTransformer.deserializeNBT(IValueTransformer.stripLdlibWrapper(t), null, null))
                    .forEach(current::add);
```
Instead of collecting the stream, then adding the collection to the list, you can just directly add each stream result to the list.

n.b. it would probably be better just to do a `forEach` or a regular loop over the listtag rather than streaming it, since you're only really doing one operation